pkg/controllers/testing/client.go

```diff
@@ -258,11 +260,58 @@ func (w *statusWriterWrapper) Patch(ctx context.Context, obj runtime.Object, pat
 //      // Makes calls to create stream return an error.
 //      rifftesting.InduceFailure("create", "Stream"),
 //   },
-func InduceFailure(verb, resource string) ReactionFunc {
+func InduceFailure(verb, kind string, o ...InduceFailureOpts) ReactionFunc {
+	opts := InduceFailureOpts{}
+	if len(o) > 0 {
+		opts = o[0]
+	}
 	return func(action Action) (handled bool, ret runtime.Object, err error) {
-		if !action.Matches(verb, resource) {
+		if !action.Matches(verb, kind) {
+			return false, nil, nil
+		}
+		if opts.Namespace != "" && opts.Namespace != action.GetNamespace() {
+			return false, nil, nil
+		}
+		if opts.Name != "" {
+			switch a := action.(type) {
+			case GetAction:
+				if opts.Name == a.GetName() {
```

Comment:
```suggestion
				if opts.Name != a.GetName() {
```
I think all the name checks be `!=`